### PR TITLE
mptcpd: add package

### PR DESCRIPTION
**Pull request by dangowrt**
Maintainer: me
Compile tested: aarch64/cortex-a53
Run tested: -

Description:
Package the Multipath TCP daemon as well as the Embedded Linux Library (ell) it depends on.
Loosely based on prior work of @jeff277

Requires openwrt/openwrt#16786

### CODE REVIEW

**Comment by vortexilation** [comment]
Thanks but I have failed to see on how ell will take over some parts of libc related functionalities.
Is it automatically work on it's own ?.
Isn't it ell is 'only' a shared library?, or do we need application which also supports ell lib?.

**Comment by dangowrt** [comment]
`ell` is simply a shared library providing *additional* functions as well as  *alternative* implementations (under different names).
`mptcpd` and `mptcpize` uses some of the functions provided by `ell`.

**Comment by vortexilation** [comment]
getting error (x86_64, glibc, gcc 14.2.0) : 
```
_dir/toolchain-x86_64_gcc-14.2.0_glibc/usr/include -I/home/user/works/openwrt/staging_dir/toolchain-x86_64_gcc-14.2.0_glibc/include -I/home/user/works/openwrt/staging_dir/target-x86_64_glibc_custom/usr/include -Wall -mtune=tigerlake -march=tigerlake -O3 -funsafe-math-optimizations -ffp-contract=fast -ftree-vectorize -fno-caller-saves -fno-plt -mtune=tigerlake -march=tigerlake -O3 -funsafe-math-optimizations -ffp-contract=fast -ftree-vectorize -fhonour-copts -fmacro-prefix-map=/home/user/works/openwrt/build_dir/target-x86_64_glibc_custom/mptcpd-0.12=mptcpd-0.12 -fstack-protector -D_FORTIFY_SOURCE=1 -O2 -fvisibility=hidden -Wextra -Werror -pedantic -fstack-protector-strong -c path_manager.c  -fPIC -DPIC -o .libs/libmptcpd_la-path_manager.o
In file included from path_manager.c:21:
/home/user/works/openwrt/staging_dir/target-x86_64_glibc_custom/usr/include/ell/genl.h: In function 'l_genl_attr_next':
/home/user/works/openwrt/staging_dir/target-x86_64_glibc_custom/usr/include/ell/genl.h:98:16: error: implicit declaration of function 'l_netlink_attr_next'; did you mean 'l_genl_attr_next'? [-Wimplicit-function-declaration]
   98 |         return l_netlink_attr_next((struct l_netlink_attr *) attr,
      |                ^~~~~~~~~~~~~~~~~~~
      |                l_genl_attr_next
/home/user/works/openwrt/staging_dir/target-x86_64_glibc_custom/usr/include/ell/genl.h: In function 'l_genl_attr_recurse':
/home/user/works/openwrt/staging_dir/target-x86_64_glibc_custom/usr/include/ell/genl.h:105:16: error: implicit declaration of function 'l_netlink_attr_recurse'; did you mean 'l_genl_attr_recurse'? [-Wimplicit-function-declaration]
  105 |         return l_netlink_attr_recurse((struct l_netlink_attr *) attr,
      |                ^~~~~~~~~~~~~~~~~~~~~~
      |                l_genl_attr_recurse
make[4]: *** [Makefile:597: libmptcpd_la-path_manager.lo] Error 1
```

**Comment by dangowrt** [comment]
Ok, thanks for testing with glibc. What does `_custom` stand for in your toolchain path?

**Comment by vortexilation** [comment]
It's only the folder named from CONFIG_BUILD_SUFFIX . Normally i am just testing around with glibc trunk/main/master version/2.41, seems faster by general on 2.41.

**Comment by vortexilation** [comment]
Based on the changelog, they have moved out the function declaration into other header file "netlink.h", the fix is simple : 
```
diff -Naur a/ell/genl.h b/ell/genl.h
--- a/ell/genl.h
+++ b/ell/genl.h
@@ -11,6 +11,7 @@
 #include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
+#include "netlink.h"
 
 #ifdef __cplusplus
 extern "C" {
```

**Comment by Neustradamus** [comment]
@dangowrt: Nice PR, thanks!

**Comment by dangowrt** [comment]
@vortexilation I managed to reproduce the problem with GCC-14 and included your fix. Please also post your patch to the ell@lists.linux.dev mailing list for inclusion in future versions. See https://subspace.kernel.org/lists.linux.dev.html on how to join the mailing lists and post there. Thank you!

**Comment by matttbe** [comment]
Thank you for packaging ell and mptcpd (and enabling MPTCP in the kernel config!)

@vortexilation @dangowrt : Regarding the error with GCC 14, it looks like the issue comes from mptcpd, because it was including only the header files it needed. Even if that sounds like the proper thing to do, that's apparently not what is supposed to be done with ell. See https://github.com/multipath-tcp/mptcpd/pull/309

In other words, no need to send a patch to ELL devs I suppose. (Or send a patch to have a proper documentation for ELL ;-) )
